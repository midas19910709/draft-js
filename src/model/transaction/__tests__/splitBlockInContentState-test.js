/**
 * Copyright (c) 2013-present, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 *
 * @emails oncall+ui_infra
 * @format
 */

'use strict';

jest.disableAutomock();

jest.mock('generateRandomKey');

const getSampleStateForTesting = require('getSampleStateForTesting');
const splitBlockInContentState = require('splitBlockInContentState');

const {contentState, selectionState} = getSampleStateForTesting();

const assertSplitBlockInContentState = selection => {
  expect(
    splitBlockInContentState(contentState, selection)
      .getBlockMap()
      .toJS(),
  ).toMatchSnapshot();
};

test('must be restricted to collapsed selections', () => {
  expect(() => {
    const nonCollapsed = selectionState.set('focusOffset', 1);
    return splitBlockInContentState(contentState, nonCollapsed);
  }).toThrow();

  expect(() => {
    return splitBlockInContentState(contentState, selectionState);
  }).not.toThrow();
});

test('must split at the beginning of a block', () => {
  assertSplitBlockInContentState(selectionState);
});

test('must split within a block', () => {
  const SPLIT_OFFSET = 3;

  assertSplitBlockInContentState(
    selectionState.merge({
      anchorOffset: SPLIT_OFFSET,
      focusOffset: SPLIT_OFFSET,
    }),
  );
});

test('must split at the end of a block', () => {
  const SPLIT_OFFSET = contentState
    .getBlockMap()
    .first()
    .getLength();

  assertSplitBlockInContentState(
    selectionState.merge({
      anchorOffset: SPLIT_OFFSET,
      focusOffset: SPLIT_OFFSET,
    }),
  );
});
