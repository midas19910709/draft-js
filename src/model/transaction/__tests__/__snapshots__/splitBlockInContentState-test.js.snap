// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must split at the beginning of a block 1`] = `
Object {
  "a": Object {
    "characterList": Array [],
    "data": Object {},
    "depth": 0,
    "key": "a",
    "text": "",
    "type": "unstyled",
  },
  "b": Object {
    "characterList": Array [
      Object {
        "entity": "1",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "1",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "1",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "1",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "1",
        "style": Array [
          "BOLD",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "b",
    "text": "Bravo",
    "type": "unordered-list-item",
  },
  "c": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "c",
    "text": "Test",
    "type": "code-block",
  },
  "d": Object {
    "characterList": Array [],
    "data": Object {},
    "depth": 0,
    "key": "d",
    "text": "",
    "type": "code-block",
  },
  "e": Object {
    "characterList": Array [],
    "data": Object {},
    "depth": 0,
    "key": "e",
    "text": "",
    "type": "code-block",
  },
  "f": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "f",
    "text": "Charlie",
    "type": "blockquote",
  },
  "key1": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "key1",
    "text": "Alpha",
    "type": "unstyled",
  },
}
`;

exports[`must split at the end of a block 1`] = `
Object {
  "a": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "a",
    "text": "Alpha",
    "type": "unstyled",
  },
  "b": Object {
    "characterList": Array [
      Object {
        "entity": "1",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "1",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "1",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "1",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "1",
        "style": Array [
          "BOLD",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "b",
    "text": "Bravo",
    "type": "unordered-list-item",
  },
  "c": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "c",
    "text": "Test",
    "type": "code-block",
  },
  "d": Object {
    "characterList": Array [],
    "data": Object {},
    "depth": 0,
    "key": "d",
    "text": "",
    "type": "code-block",
  },
  "e": Object {
    "characterList": Array [],
    "data": Object {},
    "depth": 0,
    "key": "e",
    "text": "",
    "type": "code-block",
  },
  "f": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "f",
    "text": "Charlie",
    "type": "blockquote",
  },
  "key3": Object {
    "characterList": Array [],
    "data": Object {},
    "depth": 0,
    "key": "key3",
    "text": "",
    "type": "unstyled",
  },
}
`;

exports[`must split within a block 1`] = `
Object {
  "a": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "a",
    "text": "Alp",
    "type": "unstyled",
  },
  "b": Object {
    "characterList": Array [
      Object {
        "entity": "1",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "1",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "1",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "1",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "1",
        "style": Array [
          "BOLD",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "b",
    "text": "Bravo",
    "type": "unordered-list-item",
  },
  "c": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "c",
    "text": "Test",
    "type": "code-block",
  },
  "d": Object {
    "characterList": Array [],
    "data": Object {},
    "depth": 0,
    "key": "d",
    "text": "",
    "type": "code-block",
  },
  "e": Object {
    "characterList": Array [],
    "data": Object {},
    "depth": 0,
    "key": "e",
    "text": "",
    "type": "code-block",
  },
  "f": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "f",
    "text": "Charlie",
    "type": "blockquote",
  },
  "key2": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "key2",
    "text": "ha",
    "type": "unstyled",
  },
}
`;
